pkg/client/fake/client.go

```diff
@@ -969,7 +972,15 @@ func (sw *fakeSubResourceClient) Get(ctx context.Context, obj, subResource clien
 }
 
 func (sw *fakeSubResourceClient) Create(ctx context.Context, obj client.Object, subResource client.Object, opts ...client.SubResourceCreateOption) error {
-	panic("fakeSubResourceWriter does not support create")
+	switch subResource.(type) {
```

Comment:
Would it make sense to store "subResource string" of the SubResource call (l.936) in the fakeSubResourceClient struct and then only implement this logic here if the subResource (string) is indeed "eviction"?

Reply:
Good point, updated, PTAL